packages/instant/src/util/asset.ts

```diff
@@ -0,0 +1,53 @@
+import { AssetProxyId, ObjectMap } from '@0x/types';
+import * as _ from 'lodash';
+
+import { assetDataNetworkMapping } from '../data/asset_data_network_mapping';
+import { Asset, AssetMetaData, Network, ZeroExInstantError } from '../types';
+
+export const assetUtils = {
+    createAssetFromAssetData: (
+        assetData: string,
+        assetMetaDataMap: ObjectMap<AssetMetaData>,
+        network: Network,
+    ): Asset => {
+        return {
+            assetData,
+            metaData: assetUtils.getMetaDataOrThrow(assetData, assetMetaDataMap, network),
+        };
+    },
+    getMetaDataOrThrow: (assetData: string, metaDataMap: ObjectMap<AssetMetaData>, network: Network): AssetMetaData => {
+        let mainnetAssetData: string | undefined = assetData;
```

Comment:
it looks like this could actually contain data about another network (i.e. kovan) - is this correct?  if so, we should give a different name

Reply:
To address this I simply change the order of the assignments such that it's never a lie essentially.

Reply:
I feel like I'm missing something, so feel free to correct here: to me, the following code looks like the variable name `mainnnetAssetData` could still actually be assigned to testnet data:

```ts
        if (network !== Network.Mainnet) {
            mainnetAssetData = assetUtils.getAssociatedAssetDataIfExists(assetData, network);
        }
```

If this is the case, could we just rename to `networkAssetData`?

Reply:
Ah. So in my view metaData never changes (maybe this is not the case?), so I didn't feel like duplicating it. So instead I want to be able to take an assetData string that is specific to Kovan, find the corresponding mainnet assetData string, and then use the mainnet assetData string to access our mapping of assetDatas to metaData

Reply:
Ah, I understand now.  Cool as-is.